File: LAB3/app/src/main/java/com/example/tipper/QuizActivity.java
package com.example.tipper;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.Button;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.google.android.material.bottomsheet.BottomSheetDialog;

import java.util.ArrayList;
import java.util.Random;

public class QuizActivity extends AppCompatActivity {

    private Button optionBtn1, optionBtn2, optionBtn3, optionBtn4;
    private TextView questionAttempt, question;
    private ArrayList<ModelQuiz> modelQuiz;
    Random random;
    int score = 0, attempted = 1, pos;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_quiz);

        question = findViewById(R.id.question);
        questionAttempt = findViewById(R.id.questionAttempt);
        optionBtn1 = findViewById(R.id.optionBtn1);
        optionBtn2 = findViewById(R.id.optionBtn2);
        optionBtn3 = findViewById(R.id.optionBtn3);
        optionBtn4 = findViewById(R.id.optionBtn4);

        modelQuiz = new ArrayList<>();
        random = new Random();


        getQuestions(modelQuiz);
        pos = random.nextInt(modelQuiz.size());
        setDataView(pos);

        optionBtn1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if(modelQuiz.get(pos).getAnswer().trim().toLowerCase().equals(optionBtn1.getText().toString().trim().toLowerCase())) {
                    score ++;
                }
                attempted++;
                pos = random.nextInt(modelQuiz.size());
                setDataView(pos);
            }
        });

        optionBtn2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if(modelQuiz.get(pos).getAnswer().trim().toLowerCase().equals(optionBtn2.getText().toString().trim().toLowerCase())) {
                    score ++;
                }
                attempted++;
                pos = random.nextInt(modelQuiz.size());
                setDataView(pos);
            }
        });

        optionBtn3.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if(modelQuiz.get(pos).getAnswer().trim().toLowerCase().equals(optionBtn3.getText().toString().trim().toLowerCase())) {
                    score ++;
                }
                attempted++;
                pos = random.nextInt(modelQuiz.size());
                setDataView(pos);
            }
        });

        optionBtn4.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if(modelQuiz.get(pos).getAnswer().trim().toLowerCase().equals(optionBtn4.getText().toString().trim().toLowerCase())) {
                    score ++;
                }
                attempted++;
                pos = random.nextInt(modelQuiz.size());
                setDataView(pos);
            }
        });

    }

    private void getQuestions(ArrayList<ModelQuiz> modelQuiz) {
        modelQuiz.add(new ModelQuiz("Ile porcji zawiera szklanka makaronu?", "Pół porcji","Jedną porcję","Dwie porcję","Trzy porcję","Dwie porcję"));
        modelQuiz.add(new ModelQuiz("Które z płatków śniadaniowych są najbardziej kaloryczne?", "1 kubek płatków kukurydzianych","1 kubek płatków kukurydzianych z miodem","1 kubek muesli tropikalnego","1 kubek granoli orzechowej","1 kubek granoli orzechowej"));
        modelQuiz.add(new ModelQuiz("Które mięso zawiera najmniej tłuszczu?", "Chuda mielona wołowina","Mielone mięso indycze","Kurze udka bez skóry","Mielone mięso wieprzowe","Chuda mielona wołowina"));
        modelQuiz.add(new ModelQuiz("Co ma najwięcej witaminy A?", "Ziemniak","Marchew","Szpinak","Jarmuż","Marchew"));
        modelQuiz.add(new ModelQuiz("Ile razy dziennie powinno się jeść produkty mleczne?", "2 porcje dziennie, najlepiej rano na śniadanie","2-3 razy dziennie, w ciągu całego dnia","Ludzie aktywni potrzebują od 3 do 4 porcji dziennie","Przynajmniej raz dziennie jedną porcje","2-3 razy dziennie, w ciągu całego dnia"));
        modelQuiz.add(new ModelQuiz("Jaką ilość płynów powinien wypijać codziennie aktywny człowiek?", "Mężczyźni potrzebują 330 ml (puszka), a kobiety 250 ml (szklanka) płynów","Osiem szklanek dziennie","Tyle, ile trzeba, żeby zaspokoić pragnienie","Sześć-osiem szklanek, wyłączając kawę i inne napoje moczopędne","Sześć-osiem szklanek, wyłączając kawę i inne napoje moczopędne"));
    }

    private void setDataView(int pos) {
        questionAttempt.setText("Question: " + attempted + "/6");

        if ( attempted == 6) {
            showScore();
        } else {
            question.setText(modelQuiz.get(pos).getQuestion());
            optionBtn1.setText(modelQuiz.get(pos).getOption1());
            optionBtn2.setText(modelQuiz.get(pos).getOption2());
            optionBtn3.setText(modelQuiz.get(pos).getOption3());
            optionBtn4.setText(modelQuiz.get(pos).getOption4());
        }
    }

    private void showScore() {
        final BottomSheetDialog bottomSheetDialog = new BottomSheetDialog(QuizActivity.this);
        View bottomScoreView = LayoutInflater.from(getApplicationContext()).inflate(R.layout.activity_score,(LinearLayout) findViewById(R.id.scoreView));
        TextView scor = bottomScoreView.findViewById(R.id.scores);
        Button restarts = bottomScoreView.findViewById(R.id.restart);

        scor.setText("Score: " + score + "/6");
        restarts.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                pos = random.nextInt(modelQuiz.size());
                setDataView(pos);
                attempted = 1;
                score = 0;
                bottomSheetDialog.dismiss();
            }
        });
        bottomSheetDialog.setCancelable(false);
        bottomSheetDialog.setContentView(bottomScoreView);
        bottomSheetDialog.show();
    }
}

File: LAB4/app/src/main/java/com/example/myapplicationpamo/QuizActivity.kt
package com.example.myapplicationpamo

import android.annotation.SuppressLint
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.widget.Button
import android.widget.TextView
import com.google.android.material.bottomsheet.BottomSheetDialog
import kotlinx.android.synthetic.main.activity_quiz.*
import kotlin.random.Random

class QuizActivity : AppCompatActivity() {

    private lateinit var modelQuiz: ArrayList<ModelQuiz>
    lateinit var random: Random
    var score: Int = 0
    var attempted: Int = 1
    var pos: Int = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_quiz)

        val question = findViewById<TextView>(R.id.question)
        val questionAttempt = findViewById<TextView>(R.id.questionAttempt)
        val optionBtn1 = findViewById<Button>(R.id.optionBtn1)
        val optionBtn2 = findViewById<Button>(R.id.optionBtn2)
        val optionBtn3 = findViewById<Button>(R.id.optionBtn3)
        val optionBtn4 = findViewById<Button>(R.id.optionBtn4)

        modelQuiz = ArrayList()
        random = Random

        getQuestions(modelQuiz)
        pos = random.nextInt(modelQuiz.size)

        optionBtn1.setOnClickListener {
            if(modelQuiz[pos].answer.trim().equals(optionBtn1.text.toString().trim(), ignoreCase = true)) {
                score++
            }
            attempted++
            pos = random.nextInt(modelQuiz.size)

        }

        optionBtn2.setOnClickListener {
            if(modelQuiz[pos].answer.trim().equals(optionBtn2.text.toString().trim(), ignoreCase = true)) {
                score++
            }
            attempted++
            pos = random.nextInt(modelQuiz.size)

        }

        optionBtn3.setOnClickListener {
            if(modelQuiz[pos].answer.trim().equals(optionBtn3.text.toString().trim(), ignoreCase = true)) {
                score++
            }
            attempted++
            pos = random.nextInt(modelQuiz.size)

        }

        optionBtn4.setOnClickListener {
            if(modelQuiz[pos].answer.trim().equals(optionBtn4.text.toString().trim(), ignoreCase = true)) {
                score++
            }
            attempted++
            pos = random.nextInt(modelQuiz.size)

        }
    }

    private fun getQuestions(modelQuiz: ArrayList<ModelQuiz>) {
        modelQuiz.add(ModelQuiz("Ile porcji zawiera szklanka makaronu?", "Pół porcji","Jedną porcję","Dwie porcję","Trzy porcję","Dwie porcję"))
        modelQuiz.add(ModelQuiz("Ile porcji zawiera szklanka makaronu?", "Pół porcji","Jedną porcję","Dwie porcję","Trzy porcję","Dwie porcję"))
        modelQuiz.add(ModelQuiz("Ile porcji zawiera szklanka makaronu?", "Pół porcji","Jedną porcję","Dwie porcję","Trzy porcję","Dwie porcję"))
        modelQuiz.add(ModelQuiz("Ile porcji zawiera szklanka makaronu?", "Pół porcji","Jedną porcję","Dwie porcję","Trzy porcję","Dwie porcję"))
        modelQuiz.add(ModelQuiz("Ile porcji zawiera szklanka makaronu?", "Pół porcji","Jedną porcję","Dwie porcję","Trzy porcję","Dwie porcję"))
        modelQuiz.add(ModelQuiz("Ile porcji zawiera szklanka makaronu?", "Pół porcji","Jedną porcję","Dwie porcję","Trzy porcję","Dwie porcję"))
    }

    private fun setDataView(pos: Int) {
        "Question: $attempted/6".also { questionAttempt.text = it }
        if (attempted == 6) {
            showScore()
        } else {
            question.text = modelQuiz[pos].question
            optionBtn1.text = modelQuiz[pos].option1
            optionBtn2.text = modelQuiz[pos].option2
            optionBtn3.text = modelQuiz[pos].option3
            optionBtn4.text = modelQuiz[pos].option4
        }
    }

    @SuppressLint("SetTextI18n")
    private fun showScore() {
        val bottomSheetDialog = BottomSheetDialog(this@QuizActivity)
        val bottomScoreView: View = LayoutInflater.from(getApplicationContext())
            .inflate(R.layout.activity_score, findViewById(R.id.scores))
        val scor: TextView = bottomScoreView.findViewById(R.id.scores)
        val restarts: Button = bottomScoreView.findViewById(R.id.restart)
        scor.text = "Score: $score/6"
        restarts.setOnClickListener {
            pos = random.nextInt(modelQuiz.size)
            setDataView(pos)
            attempted = 1
            score = 0
            bottomSheetDialog.dismiss()
        }
        bottomSheetDialog.setCancelable(false)
        bottomSheetDialog.setContentView(bottomScoreView)
        bottomSheetDialog.show()
    }
}